src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -30,8 +31,66 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         /// </summary>
         internal class TemporaryStorageService : ITemporaryStorageService2
         {
+            /// <summary>
+            /// The maximum size in bytes of a single storage unit in a memory mapped file which is shared with other
+            /// storage units.
+            /// </summary>
+            /// <remarks>
+            /// <para>This value was arbitrarily chosen and appears to work well. Can be changed if data suggests
+            /// something better.</para>
+            /// </remarks>
+            private const long SingleFileThreshold = 128 * 1024;
+
+            /// <summary>
+            /// The size in bytes of a memory mapped file created to store multiple temporary objects.
+            /// </summary>
+            /// <remarks>
+            /// <para>This value was arbitrarily chosen and appears to work well. Can be changed if data suggests
+            /// something better.</para>
+            /// </remarks>
+            private const long MultiFileBlockSize = SingleFileThreshold * 32;
+
             private readonly ITextFactoryService _textFactory;
 
+            /// <summary>
+            /// The synchronization object for accessing the memory mapped file related fields.
+            /// </summary>
+            /// <remarks>
+            /// <para>PERF DEV NOTE: A concurrent (but complex) implementation of this type with identical semantics is
+            /// available in source control history. The use of exclusive locks was not causing any measurable
+            /// performance overhead even on 28-thread machines at the time this was written.</para>
+            /// </remarks>
+            private readonly object _gate = new object();
+
+            /// <summary>
+            /// The most recent memory mapped file for creating multiple storage units. It will be used via bump-pointer
+            /// allocation until space is no longer available in it.
+            /// </summary>
+            /// <remarks>
+            /// <para>Access should be synchronized on <see cref="_gate"/>.</para>
+            /// </remarks>
+            private ReferenceCountedDisposable<MemoryMappedFile>.WeakReference _weakFileReference;
+
+            /// <summary>The name of the current memory mapped file for multiple storage units.</summary>
```

Comment:
In all cases "the current memory mapped file" is the thing referenced by _weakFileReference? Might be good to cref it so the relationship is explicit.